### PR TITLE
:bug: Do NOT throw an error on `denops#server#close`

### LINKED ISSUE #337: `denops#server#close()` throws an error if channel does not exist
`denops#server#close()` throws an error if the channel does not yet exist.
But it is expected that it should not throw an error.

Document says following:
https://github.com/vim-denops/denops.vim/blob/c727a3f05a804d1f04b11735f8241518465de9a7/doc/denops.txt#L237-L240

## Actual

Call `denops#server#close()` then outputs:

```
Error detected while processing function denops#server#close[1]..denops#_internal#server#chan#close[2]..function denops#server#close[1]..denops#_internal#server#chan#close:
line    2:
E605: Exception not caught: [denops] Channel does not exist yet
```

## Expected

No errors occurred.

## Additional information

Similar item `denops#server#stop()` is also says "It does nothing when ...":
https://github.com/vim-denops/denops.vim/blob/c727a3f05a804d1f04b11735f8241518465de9a7/doc/denops.txt#L215-L218

And it doesn't throw any errors.

### PR DESCRIPTION
**Pull request by lambdalisue**
Close #337

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved error handling in channel operations by adding a debug message when attempting to close a non-existent channel.
	- Simplified logic in `denops#server#restart()` and `denops#server#reconnect()` functions for better control flow.

### CODE REVIEW

**Review by Shougo — APPROVED** [review]

File: autoload/denops/_internal/server/chan.vim [hunk 1]
```diff
@@ -60,7 +60,8 @@ endfunction
 
 function! denops#_internal#server#chan#close() abort
   if s:chan is# v:null
-    throw '[denops] Channel does not exist yet'
+    call denops#_internal#echo#debug("Channel does not exist yet")
+    return
   endif
```
Comment (Milly) [inline autoload/denops/_internal/server/chan.vim:65]: `denops#_internal#server#chan#close()` is an internal method, so throwing it should let us catch the bug.
No changes should be made here.

**Review by Milly — CHANGES_REQUESTED** [review]
Where `denops#_internal#server#chan#close()` is used, I suggest surrounding it with `if denops#_internal#server#chan#is_connected() | ... | endif`.

File: autoload/denops/_internal/server/chan.vim (hunk 1 above)
Comment (lambdalisue) [reply autoload/denops/_internal/server/chan.vim:65]: Agree.

Comment (Milly) [reply autoload/denops/_internal/server/chan.vim:65]: Add reverse patch or delete commit 3170593.

**Review by Milly — CHANGES_REQUESTED** [review]
Unresolved first review point.
Keep `throw`.

**Comment by lambdalisue** [comment]
Oops. Rebase failure.

**Review by Milly — APPROVED** [review]

**Pull request merged**